src/block-ref.test.ts

```diff
@@ -106,11 +106,7 @@ describe('createBlockRefMiddleware', () => {
                         '0x100',
                       ),
                     },
-                    response: (req) => ({
-                      id: req.id,
-                      jsonrpc: '2.0' as const,
-                      result: 'something',
-                    }),
+                    response: async () => Promise.resolve('something'),
```

Comment:
Nit: You don't need to wrap the return value of an `async` function in a promise, it will do that for you. So this should be enough:

``` suggestion
                    response: async () => 'something',
```

(I won't note the other instances of this, but you get the idea)